### DISCUSSION TITLE
Reloading custom fabricJS canvas objects from JSON?

### DISCUSSION THREAD

**Discussion by Milou6** [fabricjs/fabric.js#6978] [Q&A] [answered]
Hello all,
I have been stuck on an issue trying to reload custom fabricJS objects from JSON, and after extensive search and a detailed question on stackoverflow (posted almost 2 weeks ago), still stuck on it.

The original question is [right here](https://stackoverflow.com/questions/66711085/how-do-i-reload-custom-fabricjs-canvas-objects-from-json). 

As I mentioned on the question, I added non-minified imports of fabricJS to be able to see exactly where in the source-code an error occurs. But ideally, I'd like to understand what am I doing wrong without having to play with source code.

_Any help_, even pointing me to someone who could help me with this, is highly appreciated.
Thank you all for your work on this awesome project!

**Comment by asturur** [comment, marked as answer]
Your code has an issue, is not following the polyline structure.
A polyline starts like this:

```new fabric.Polyline(points, options)```

And so the fromObject for the polyline specifically is:

```
  fabric.Polyline.fromObject = function(object, callback) {
    return fabric.Object._fromObject('Polyline', object, callback, 'points');
  };
```

Try to have the correct initialize function and a fromObject function that refer to points on the last argument, i assume it could fix your issues

**Reply by Milou6** [reply]
@asturur Really appreciate your help. The fix works great! 😁
If that's okay with you, I'll post your answer on my original SO question, so maybe it helps someone else in the future.

**Reply by asturur** [reply]
Sure, the more people read the better.
